view.php

```diff
@@ -20,32 +20,46 @@
 require_once($CFG->libdir.'/tablelib.php');
 
 $id = required_param('id', PARAM_INT);
+$download = optional_param('download', false, PARAM_ALPHA);
 
 list ($course, $cm) = get_course_and_cm_from_cmid($id, 'cpdlogbook');
 
 require_course_login($course, false, $cm);
 
 $record = $DB->get_record('cpdlogbook', [ 'id' => $cm->instance ], '*', MUST_EXIST);
 
+// If the table is being downloaded, then the $actions parameter is set to false.
+$table = new entries_table($cm, $USER->id, $OUTPUT, !$download, 'cpdlogbook_id');
+
+$table->is_downloading($download, 'cpdlogbook'.time(), 'cpdlogbook');
```

Comment:
Perhaps a better file name could be the name of the module instance followed by the persons name and the date, eg

'MyCPD Brendan Heywood 2021-08-22.csv'

To format the date use userdate() and for the format string use one of these:

https://github.com/moodle/moodle/blob/master/lang/en/langconfig.php#L40-L42

https://docs.moodle.org/dev/Time_API